client/src/main/java/org/eclipse/hono/client/CommandResponse.java

```diff
@@ -88,29 +96,39 @@ public static CommandResponse from(
     }
 
     /**
-     * Creates a command response for a reply-to ID, correlation ID and status.
+     * Creates a command response from a given message.
      *
-     * @param tenantId The tenant ID of the device sending the response.
-     * @param deviceId The device ID of the device sending the response.
-     * @param payload The command response payload.
-     * @param contentType The content-type of the response payload.
-     * @param status The HTTP status code indicating the outcome of executing the command on the device.
-     * @param correlationId The identifier used to correlate this response with the command request.
-     * @param address The address that the command response is to be sent to.
-     * 
-     * @return The command response or {@code null} if any of correlationId, replyTo and status is null or if the
-     *         status code is &lt; 200 or &gt;= 600.
+     * @param message The command response message.
+     *
+     * @return The command response or {@code null} if message or any of correlationId, address and status is null or if
+     *         the status code is &lt; 200 or &gt;= 600.
      */
-    public static CommandResponse from(final String tenantId, final String deviceId, final Buffer payload,
-            final String contentType, final Integer status,
-            final String correlationId, final ResourceIdentifier address) {
-        if (correlationId == null || address == null || status == null) {
+    public static CommandResponse from(final Message message) {
+        if (message == null) {
+            return null;
+        }
+
+        final String correlationId = Optional.ofNullable(message.getCorrelationId())
+                .map(id -> {
+                    if (id instanceof String) {
+                        return (String) id;
+                    } else {
+                        return null;
+                    }
+                }).orElse(null);
+
+        final Integer status = MessageHelper.getApplicationProperty(message.getApplicationProperties(),
+                MessageHelper.APP_PROPERTY_STATUS, Integer.class);
+
+        if (correlationId == null || message.getAddress() == null || status == null) {
             return null;
         } else if (INVALID_STATUS_CODE.test(status)) {
             return null;
         } else {
-            return new CommandResponse(tenantId, deviceId, payload, contentType, status, correlationId,
-                    getReplyId(deviceId, address));
+            final ResourceIdentifier resource = ResourceIdentifier.fromString(message.getAddress());
```

Comment:
> I wonder if it would be helpful to replace the fields of `CommandResponse` with a `Message` for storing the properties. In the end, producing a `Message` to be sent to an application is the whole purpose of the `CommandResponse` class. That way we would also prevent the reading of properties from a response AMQP message received from a device just in order to later putting them into a new AMP message to be sent to the application. WDYT?

@sophokles73 Currently CommandResponseSender is responsible for creating a message out of the `CommandResponse` fields. I like the idea to replace the existing fields with a `Message`.
I would suggest a separate PR to replace the fields with a `Message`. I can raise it as a 3rd PR. WDYT?

Reply:
> this can throw a runtime exception

@sophokles73 The `NullPointerException` and  `IllegalArgumentException` that may be thrown by `ResourceIdentifier.fromString(...` have been handled and the changes are pushed.